commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java

```diff
@@ -179,6 +185,20 @@ public static ExecutorService newCachedThreadPoolWithMaxThreads(int maxThreads,
                 .build();
     }
 
+    @Beta
+    public static ExecutorService newCachedThreadPoolWithMaxThreadsNoSpan(int maxThreads, String name) {
+        Preconditions.checkNotNull(name, "Name is required");
+        Preconditions.checkArgument(!name.isEmpty(), "Name must not be empty");
+        Preconditions.checkArgument(maxThreads > 0, "Max threads must be positive");
+        return MetricRegistries.executor()
+                .registry(SharedTaggedMetricRegistries.getSingleton())
+                .name(name)
+                .executor(PTExecutors.wrap(getViewExecutor(name, maxThreads, 0, SHARED_EXECUTOR.get())))
```

Comment:
Another way to do this instead of creating parallel code would be to make the code use optionals instead and just make the final wrap method handle the difference.

This felt better to me, but happy to be convinced otherwise.